### PR TITLE
Add cached property to the response

### PR DESCRIPTION
**Pull request by mtzfederico**
This PR adds a cached property to the response. This lets you check if the response was cached or not.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +161,13 @@ The IPinfo library can be authenticated with your IPinfo API token, which is pas
 
 In-memory caching of `details` data is provided by default via the [cachetools](https://cachetools.readthedocs.io/en/latest/) library. This uses an LRU (least recently used) cache with a TTL (time to live) by default. This means that values will be cached for the specified duration; if the cache's max size is reached, cache values will be invalidated as necessary, starting with the oldest cached value.
 
+You can access the cached property to see if the response was cached.
```
Comment (UmanShahzad) [inline README.md:164]: Two nits here:

1. Add a newline between the codeblock and this line.
2. Convert the word "cached" here to an inline codeblock, i.e. "You can access the `cached` property..." This clearly distinguishes it as referring to something in the code.

File: ipinfo/handler.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
     cache_key,
 )
 from . import handler_utils
+from ipinfo import details
```
Comment (UmanShahzad) [inline ipinfo/handler.py:27]: What is this import?

**Review by UmanShahzad — COMMENTED** [review]
@mtzfederico thank you for the PR. I'd like to actually know what use case this serves, because currently it's not clear to me. I would imagine that if it needs to be known that a certain IP is in the cache or not, it'd be better to use the cache interface directly to do that.

**Comment by mtzfederico** [comment]
I find it useful for some simple statistics and for debugging, but you mention that it would be easier to use the cache interface. I am not very familiar with cachetools,  it might be easier to just add some sample code for debugging to the docs.

I am not sure about that import, I just ran my tests again without it and it works. My guess is that my editor added it automatically.

File: README.md (hunk 1 above)
Comment (mtzfederico) [reply README.md:164]: I accidentally clicked resolve. I'm working on this.

File: README.md [hunk 3]
```diff
@@ -161,6 +161,14 @@ The IPinfo library can be authenticated with your IPinfo API token, which is pas
 
 In-memory caching of `details` data is provided by default via the [cachetools](https://cachetools.readthedocs.io/en/latest/) library. This uses an LRU (least recently used) cache with a TTL (time to live) by default. This means that values will be cached for the specified duration; if the cache's max size is reached, cache values will be invalidated as necessary, starting with the oldest cached value.
 
+You can access the cached property to see if the response was `cached`.
```
Comment (UmanShahzad) [inline README.md:164]: The wrong `cached` was wrapped with '`'; in the example at https://github.com/ipinfo/python/pull/54#discussion_r664774261 I wrapped only the first one.

**Comment by UmanShahzad** [comment]
@mtzfederico You wouldn't have to work directly with `cachetools`, because that's an implementation of the cache interface, which can be found at https://github.com/ipinfo/python/blob/master/ipinfo/cache/interface.py.

You can access/update the cache via dictionary-like notation, but the cache key must be gotten via `ipinfo.handler_utils.cache_key`. For example:

```python
>>> import ipinfo
>>> from ipinfo.handler_utils import cache_key
>>> h = ipinfo.getHandler()
>>> ip_cache_key = cache_key('1.1.1.1')

# __contains__
>>> ip_cache_key in h.cache

# __getitem__
>>> handler.cache[ip_cache_key]

# __setitem__
>>> h.cache[ip_cache_key] = None
```

If this is sufficient I'd much rather we just make a PR explaining this, rather than adding a new field which is really only metadata.

**Comment by mtzfederico** [comment]
@UmanShahzad Thanks for your examples. I'm going to close this PR and create another one with the examples that you provided. Should I put it in the README.md below the "Using a different cache" section?

**Comment by UmanShahzad** [comment]
@mtzfederico that would be great, thank you. Yes you can make a new section below that as:

```md
#### Accessing the cache directly

...
```

**Pull request closed without merging**